tests/next_tests/unit_tests/iterator_tests/transforms_tests/test_cse.py

```diff
@@ -133,22 +134,45 @@ def common_expr():
     assert actual == expected
 
 
-def test_if_can_deref():
-    """
-    Test no subexpression is moved outside expressions of the form `if_(can_deref(...), ..., ...)`
-    """
-    # if can_deref(⟪Iₒ, 1ₒ⟫(it)) then ·⟪Iₒ, 1ₒ⟫(it) else ·⟪Iₒ, 1ₒ⟫(it) + ·⟪Iₒ, 1ₒ⟫(it)
+def test_if_can_deref_no_extraction():
```

Comment:
Related to one of my questions above, if conditional expressions other than `if can_deref` are allowed, there should be some test for it